src/java/htsjdk/samtools/CRAMFileReader.java

```diff
@@ -491,68 +445,47 @@ void enableFileSource(final SamReader reader, final boolean enabled) {
             iterator.setFileSource(enabled ? reader : null);
     }
 
-    private class MultiIntervalIterator implements SAMRecordIterator {
-        private final Iterator<QueryInterval> queries;
-        private CloseableIterator<SAMRecord> iterator;
-        private final boolean overlap;
+    private class CRAMIntervalIterator
+            extends BAMQueryMultipleIntervalsIteratorFilter
+            implements SAMRecordIterator {
 
-        public MultiIntervalIterator(final Iterator<QueryInterval> queries, final boolean overlap) {
-            this.queries = queries;
-            this.overlap = overlap;
-        }
+        private CRAMIterator unfilteredIterator;
+        SAMRecord nextRec = null;
+        boolean hasMore = false;
 
-        @Override
-        public SAMRecordIterator assertSorted(final SortOrder sortOrder) {
-            return null;
-        }
-
-        @Override
-        public void close() {
-
-        }
-
-        @Override
-        public boolean hasNext() {
-            if (iterator == null || !iterator.hasNext()) {
-                if (!queries.hasNext()) return false;
-                do {
-                    final QueryInterval query = queries.next();
-                    try {
-                        iterator = query(query.referenceIndex, query.start, query.end, overlap);
-                    } catch (final IOException e) {
-                        throw new RuntimeException(e);
-                    }
-                } while (!iterator.hasNext() && queries.hasNext());
+        public CRAMIntervalIterator(final QueryInterval[] queries, final boolean contained) {
+            super(queries, contained);
+            if (queries.length== 0) {
+                throw new IllegalArgumentException("Empty interval list passed to interval iterator");
             }
-            return iterator.hasNext();
-        }
 
-        @Override
-        public SAMRecord next() {
-            return iterator.next();
-        }
-
-        @Override
-        public void remove() {
-            iterator.remove();
+            long[] coordinates = coordinatesFromQueryIntervals(getIndex(), queries);
+            if (coordinates.length != 0) {
+                hasMore = true;
```

Comment:
I don't actually know if thats possible (I would not expect it to happen - if it did I think it would be a bug in the BAMFileSpan code). However the iterator filter should filter those records out in either case for.